llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp

```diff
@@ -4159,6 +4172,13 @@ bool X86AsmParser::matchAndEmitATTInstruction(
     SwitchMode(X86::Is16Bit);
     ForcedDataPrefix = 0;
   }
+
+  // When "-msse2avx" option is enabled replaceSSE2AVXOpcode method will
+  // replace SSE instruction with equivalent AVX instruction using mapping given
+  // in table GET_X86_SSE2AVX_TABLE
+  if (MCOptions.SSE2AVX)
```

Comment:
Not duplicate it in ATT and INTEL. Move it to `X86AsmParser::processInstruction`

Reply:
Done